pkg/bindings/connection.go

```diff
@@ -140,19 +145,45 @@ func NewConnectionWithIdentity(ctx context.Context, uri string, identity string,
 	return ctx, nil
 }
 
-func tcpClient(_url *url.URL) Connection {
+func tcpClient(_url *url.URL) (Connection, error) {
 	connection := Connection{
 		URI: _url,
 	}
+	dialContext := func(ctx context.Context, _, _ string) (net.Conn, error) {
+		return net.Dial("tcp", _url.Host)
+	}
+	// use proxy if env `CONTAINER_PROXY` set
+	if proxyURI, found := os.LookupEnv("CONTAINER_PROXY"); found {
+		proxyURL, err := url.Parse(proxyURI)
+		if err != nil {
+			return connection, fmt.Errorf("value of CONTAINER_PROXY is not a valid url: %s: %w", proxyURI, err)
+		}
+		proxyDialer, err := proxy.FromURL(proxyURL, proxy.Direct)
+		if err != nil {
+			return connection, fmt.Errorf("unable to dial to proxy %s, %w", proxyURI, err)
+		}
+		dialContext = func(ctx context.Context, _, _ string) (net.Conn, error) {
+			logrus.Debugf("use proxy %s, but proxy dialer does not support dial timeout", proxyURI)
+			return proxyDialer.Dial("tcp", _url.Host)
+		}
+		if f, ok := proxyDialer.(proxy.ContextDialer); ok {
+			dialContext = func(ctx context.Context, _, _ string) (net.Conn, error) {
+				// the default tcp dial timeout seems to be 75s, podman-remote will retry 3 times bofere exit.
+				// here we change proxy dial timeout to 3s
+				logrus.Debugf("use proxy %s with dial timeout 3s", proxyURI)
+				ctx, unused := context.WithTimeout(ctx, time.Second*3)
+				_ = unused // for govet golangci-lint
```

Comment:
can be written as AFAIK
```suggestion
				ctx, _ := context.WithTimeout(ctx, time.Second*3)
```

Reply:
the podman-remote man page  is `docs/source/markdown/podman-remote.1.md` ?

Reply:
Yes